packages/admin/cms-admin/src/dam/blocks/DamFileDownloadLinkBlock.tsx

```diff
@@ -0,0 +1,141 @@
+            fileUrl: damFile.fileUrl,
+        };
+
+        return ret;
+    },
+
+    definesOwnPadding: true,
+
+    AdminComponent: ({ state, updateState }) => {
+        return (
+            <>
+                <BlocksFinalForm<{
+                    openFileType: DamFileDownloadLinkBlockData["openFileType"];
+                    file?: DamFileDownloadLinkBlockData["file"];
+                }>
+                    onSubmit={(newValues) => {
+                        updateState({
+                            file: newValues.file ?? undefined,
+                            openFileType: newValues.openFileType,
+                        });
+                    }}
+                    initialValues={{
+                        file: state.file,
+                        openFileType: state.openFileType ?? "Download",
+                    }}
+                >
+                    {state.file === undefined ? (
+                        <Field name="file" component={FileField} fullWidth />
+                    ) : (
+                        <AdminComponentPaper disablePadding>
+                            <Box padding={3}>
+                                <Typography variant="subtitle1">{state.file.name}</Typography>
+                                <Typography variant="body1" color="textSecondary">
+                                    <DamPathLazy fileId={state.file.id} />
+                                </Typography>
+                            </Box>
+                            <Divider />
+                            <AdminComponentButton startIcon={<Delete />} onClick={() => updateState({ ...state, file: undefined })}>
+                                <FormattedMessage {...messages.empty} />
+                            </AdminComponentButton>
+                        </AdminComponentPaper>
+                    )}
+
+                    <Divider />
+                    <AdminComponentPaper>
+                        <Field
+                            name="openFileType"
+                            fullWidth
+                            label={<FormattedMessage id="blocks.damFileDownloadLink.openFileType" defaultMessage="Open file" />}
+                        >
+                            {(props) => (
+                                <>
+                                    <FinalFormSelect {...props}>
+                                        <MenuItem value={openFileTypeOptions[0].value}>{openFileTypeOptions[0].name}</MenuItem>
+                                        <MenuItem value={openFileTypeOptions[1].value}>{openFileTypeOptions[1].name}</MenuItem>
+                                    </FinalFormSelect>
+                                </>
+                            )}
+                        </Field>
+                    </AdminComponentPaper>
+                </BlocksFinalForm>
+            </>
+        );
+    },
+
+    isValid: (state) => {
+        return !!state.file;
+    },
+};
+
+const openFileTypeOptions = [
```

Comment:
I would not create a variable for this, you use it exactly once and it makes the code unnecessary complicated. At least it must be an objected instead of an array; or you map thru the array.